src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -150,146 +147,89 @@ private static void ScavengeCache()
-						adjusted
-							.Replace(
-								"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, originalPrefix),
-								"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, document.DocumentElement.GetNamespaceOfPrefix(""))
-							);
+						throw new InvalidOperationException($"Syntax error while parsing conditional namespace expression {namespaceUri}");
 					}
-					else if (indexOfUsingColon > 0 && originalPrefix[indexOfUsingColon - 1] == '#')
+
+					return new KeyValuePair<bool?, string>(methodName == nameof(ApiInformation.IsApiContractPresent) ?
+						ApiInformation.IsApiContractPresent(elements[1], majorVersion) :
+						ApiInformation.IsApiContractNotPresent(elements[1], majorVersion), valueSplit[0]);
+				case nameof(ApiInformation.IsTypePresent):
+				case nameof(ApiInformation.IsTypeNotPresent):
+					if (elements.Length < 2)
 					{
-						// We have "#using:", we want to keep it.
-						adjusted
-							.Replace(
-								"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, originalPrefix),
-								"xmlns:{0}=\"{1}\"".InvariantCultureFormat(n, originalPrefix.Substring(indexOfUsingColon - 1))
-							);
+						throw new InvalidOperationException($"Syntax error while parsing conditional namespace expression {namespaceUri}");
 					}
-				}
+					var expectedType = elements[1];
+					return new KeyValuePair<bool?, string>(methodName == nameof(ApiInformation.IsTypePresent) ?
+						ApiInformation.IsTypePresent(elements[1], _metadataHelper) :
+						ApiInformation.IsTypeNotPresent(elements[1], _metadataHelper), valueSplit[0]);
+				default:
+					return new KeyValuePair<bool?, string>(null, namespaceUri);// TODO: support IsPropertyPresent
 			}
+		}
 
-			foreach (var includedCond in conditionals.IncludedConditionals)
-			{
-				var valueSplit = includedCond.Value.Split('?');
-				// Strip the conditional part, so the namespace can be parsed correctly by the Xaml reader
-				adjusted
-					.Replace(
-						includedCond.OuterXml,
-						"{0}=\"{1}\"".InvariantCultureFormat(includedCond.Name, valueSplit[0])
-					);
-			}
+		private XmlReader ApplyIgnorables(SourceText sourceText)
+		{
+			var originalString = sourceText.ToString();
+
+			var document = new XmlDocument();
+			document.LoadXml(originalString);
 
-			if (hasxBind)
+			//var conditionals = FindConditionals(document); /*TODO*/
+
+			var hasxBind = originalString.Contains("{x:Bind", StringComparison.Ordinal);
+
+			if (!hasxBind)
 			{
-				// Apply replacements to avoid having issues with the XAML parser which does not
-				// support quotes in positional markup extensions parameters.
-				// Note that the UWP preprocessor does not need to apply those replacements as the
-				// x:Bind expressions are being removed during the first phase and replaced by "connections".
-				adjusted = new(XBindExpressionParser.RewriteDocumentPaths(adjusted.ToString()));
+				// No need to modify file
+				return XmlReader.Create(new StringReader(originalString));
```

Comment:
NOTE TO SELF: originalString is declared at the top with `sourceText.ToString()`. We should see if a stream-based approach can work to avoid potential costly allocations.

Reply:
I'll revisit later if this appeared in traces.